apps/aecore/src/aec_sync.erl

```diff
@@ -913,3 +933,40 @@ sync_progress(#state{sync_tasks = SyncTasks} = State) ->
                 false -> SyncProgress
             end
     end.
+
+pretty_print_chain_block(#chain_block{hash = Hash, height = Height}) ->
+    #{ hash => Hash
```

Comment:
You could generalize these pretty print function by taking using the `record_fields` to dynamically build that map.

Reply:
Ahem... here is of course the ultimate record pretty-printing generalization ;-)
https://github.com/uwiger/jobs/blob/master/src/jobs_info.erl

Reply:
That is pretty much what I had in mind. Someone please submit a PR to OTP :-)

Reply:
Don't make it more complex than necessary, but please don't use function names spanning half a page...